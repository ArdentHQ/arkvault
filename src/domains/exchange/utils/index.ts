export * from "./assertions";
export * from "./errors";
