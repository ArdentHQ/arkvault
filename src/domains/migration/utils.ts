import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
