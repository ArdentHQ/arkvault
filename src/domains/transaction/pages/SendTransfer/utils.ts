export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
