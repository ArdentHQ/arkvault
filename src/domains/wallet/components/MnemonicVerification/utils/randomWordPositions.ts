import { sample } from "@ardenthq/sdk-helpers";

const randomWordPositions = (length: number): number[] => {
	const positions: number[] = [...Array.from({ length }).keys()];
	const result: number[] = [];

	while (result.length < 3) {
		const randomNumber = sample(positions) + 1;

		if (result.includes(randomNumber)) {
			continue;
		}

		result.push(randomNumber);
	}

	return result.sort((a, b) => a - b);
};

export { randomWordPositions };
