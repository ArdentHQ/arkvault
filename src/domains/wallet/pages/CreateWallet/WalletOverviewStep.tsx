import React, { useEffect } from "react";
import { useFormContext } from "react-hook-form";
import { Trans, useTranslation } from "react-i18next";

import { Alert } from "@/app/components/Alert";
import { Header } from "@/app/components/Header";
import { Toggle } from "@/app/components/Toggle";
import { toasts } from "@/app/services";
import { MnemonicList, MnemonicListSkeleton } from "@/domains/wallet/components/MnemonicList";
import { useFiles } from "@/app/hooks/use-files";
import { CopyOrDownload } from "@/app/components/CopyOrDownload";
import { ThemeIcon } from "@/app/components/Icon";
import { Divider } from "@/app/components/Divider";

export const WalletOverviewStep = ({ isGeneratingWallet }: { isGeneratingWallet: boolean }) => {
	const { getValues, setValue, unregister, watch } = useFormContext();

	const { wallet, mnemonic } = watch();

	const { isLegacy, showSaveDialog } = useFiles();

	const { t } = useTranslation();

	useEffect(() => {
		unregister("verification");
	}, [unregister]);

	const handleDownload = async () => {
		try {
			const filePath = await showSaveDialog(mnemonic, { fileName: `${wallet.address()}.txt` });

			if (!isLegacy()) {
				toasts.success(<Trans i18nKey="COMMON.SAVE_FILE.SUCCESS" values={{ filePath }} />);
			}
		} catch (error) {
			if (!error.message.includes("The user aborted a request")) {
				toasts.error(t("COMMON.SAVE_FILE.ERROR", { error: error.message }));
			}
		}
	};

	const handleToggleEncryption = (event: React.ChangeEvent<HTMLInputElement>) => {
		setValue("useEncryption", event.target.checked);
	};

	return (
		<section data-testid="CreateWallet__WalletOverviewStep">
			<Header
				title={t("WALLETS.PAGE_CREATE_WALLET.PASSPHRASE_STEP.TITLE")}
				titleIcon={
					<ThemeIcon
						darkIcon="YourPassphraseDark"
						lightIcon="YourPassphraseLight"
						greenDarkIcon="YourPassphraseDarkGreen"
						greenLightIcon="YourPassphraseLightGreen"
						dimensions={[24, 24]}
					/>
				}
				className="hidden sm:block"
			/>

			<div className="mt-4 space-y-4">
				<Alert className="mt-6">{t("WALLETS.PAGE_CREATE_WALLET.PASSPHRASE_STEP.WARNING")}</Alert>

				{isGeneratingWallet ? <MnemonicListSkeleton /> : <MnemonicList mnemonic={mnemonic} />}

				<CopyOrDownload
					title={t("WALLETS.PAGE_CREATE_WALLET.PASSPHRASE_STEP.COPY_OR_DOWNLOAD.TITLE")}
					description={t("WALLETS.PAGE_CREATE_WALLET.PASSPHRASE_STEP.COPY_OR_DOWNLOAD.DESCRIPTION")}
					copyData={mnemonic}
					onClickDownload={() => handleDownload()}
					disabled={isGeneratingWallet}
				/>

				<Divider />

				<div className="flex w-full flex-col space-y-2">
					<div className="flex items-center justify-between space-x-5">
						<span className="font-bold text-theme-secondary-text">
							{t("WALLETS.PAGE_CREATE_WALLET.PASSPHRASE_STEP.ENCRYPTION.TITLE")}
						</span>

						<span data-testid="CreateWallet__encryption">
							<Toggle
								data-testid="CreateWallet__encryption-toggle"
								defaultChecked={getValues("useEncryption")}
								onChange={handleToggleEncryption}
								disabled={isGeneratingWallet}
							/>
						</span>
					</div>

					<span className="mr-12 text-sm text-theme-secondary-500">
						{t("WALLETS.PAGE_CREATE_WALLET.PASSPHRASE_STEP.ENCRYPTION.DESCRIPTION")}
					</span>
				</div>
			</div>
		</section>
	);
};
