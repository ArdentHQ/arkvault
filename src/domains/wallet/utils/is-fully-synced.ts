import { Contracts } from "@payvo/sdk-profiles";

export const isFullySynced = (wallet: Contracts.IReadWriteWallet): boolean =>
	wallet.hasBeenFullyRestored() && (wallet.hasSyncedWithNetwork() || wallet.balance() === 0);
