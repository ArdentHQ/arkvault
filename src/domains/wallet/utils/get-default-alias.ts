import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { networkDisplayName } from "@/utils/network-utils";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${networkDisplayName(network)} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
