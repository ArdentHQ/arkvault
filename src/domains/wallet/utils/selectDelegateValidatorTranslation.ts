import { Networks } from "@ardenthq/sdk";
import { isMainsailNetwork } from "@/utils/network-utils";

interface Properties<T extends string | React.ReactNode> {
	network: Networks.Network;
	validatorStr: T;
	delegateStr: T;
}

export const selectDelegateValidatorTranslation = <T extends string | React.ReactNode>({ network, validatorStr, delegateStr }: Properties<T>): T => isMainsailNetwork(network) ? validatorStr : delegateStr;



