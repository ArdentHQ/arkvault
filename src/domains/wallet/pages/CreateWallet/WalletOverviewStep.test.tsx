/* eslint-disable @typescript-eslint/require-await */
import { Contracts } from "@ardenthq/sdk-profiles";
import { renderHook } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as browserAccess from "browser-fs-access";
import React from "react";
import { FormProvider, useForm } from "react-hook-form";
import { Trans } from "react-i18next";

import { WalletOverviewStep } from "./WalletOverviewStep";
import { toasts } from "@/app/services";
import { env, getDefaultProfileId, MNEMONICS, render, screen, waitFor } from "@/utils/testing-library";

let profile: Contracts.IProfile;

const renderForm = () =>
	renderHook(() =>
		useForm({
			defaultValues: {
				mnemonic: MNEMONICS[0],
				wallet: {
					address: () => "address",
				},
			},
		}),
	);

describe("WalletOverviewStep", () => {
	beforeEach(() => {
		profile = env.profiles().findById(getDefaultProfileId());

		for (const wallet of profile.wallets().values()) {
			profile.wallets().forget(wallet.id());
		}
	});

	describe("Render step", () => {
		it("should render", async () => {
			const { result: form } = renderForm();

			const browserAccessMock = vi
				.spyOn(browserAccess, "fileSave")
				// @ts-ignore
				.mockResolvedValue({ name: "filePath" });

			const { asFragment } = render(
				<FormProvider {...form.current}>
					<WalletOverviewStep />
				</FormProvider>,
			);

			expect(asFragment()).toMatchSnapshot();

			const writeTextMock = vi.fn();
			const clipboardOriginal = navigator.clipboard;
			// @ts-ignore
			navigator.clipboard = { writeText: writeTextMock };

			await userEvent.click(screen.getByTestId("clipboard-icon__wrapper"));

			await waitFor(() => expect(writeTextMock).toHaveBeenCalledWith(MNEMONICS[0]));

			// @ts-ignore
			navigator.clipboard = clipboardOriginal;

			browserAccessMock.mockRestore();
		});

		it("should download file if legacy", async () => {
			const { result: form } = renderForm();

			const browserAccessMock = vi
				.spyOn(browserAccess, "fileSave")
				// @ts-ignore
				.mockResolvedValue({ name: "filePath" });

			const toastSpy = vi.spyOn(toasts, "success");

			render(
				<FormProvider {...form.current}>
					<WalletOverviewStep />
				</FormProvider>,
			);

			await userEvent.click(screen.getByTestId("CopyOrDownload__download"));

			await waitFor(() => {
				expect(toastSpy).not.toHaveBeenCalledWith(
					<Trans i18nKey="COMMON.SAVE_FILE.SUCCESS" values={{ filePath: "filePath" }} />,
				);
			});

			toastSpy.mockRestore();
			browserAccessMock.mockRestore();
		});

		it("should show success toast on successful download", async () => {
			Object.defineProperty(window, "showSaveFilePicker", vi.fn());

			const { result: form } = renderForm();

			const browserAccessMock = vi
				.spyOn(browserAccess, "fileSave")
				// @ts-ignore
				.mockResolvedValue({ name: "filePath" });

			const toastSpy = vi.spyOn(toasts, "success");

			render(
				<FormProvider {...form.current}>
					<WalletOverviewStep />
				</FormProvider>,
			);

			await userEvent.click(screen.getByTestId("CopyOrDownload__download"));

			await waitFor(() => {
				expect(toastSpy).toHaveBeenCalledWith(
					<Trans i18nKey="COMMON.SAVE_FILE.SUCCESS" values={{ filePath: "filePath" }} />,
				);
			});

			toastSpy.mockRestore();
			browserAccessMock.mockRestore();
		});

		it("should not show success toast on cancelled download", async () => {
			const { result: form } = renderForm();

			const browserAccessMock = vi
				.spyOn(browserAccess, "fileSave")
				.mockRejectedValue(new Error("The user aborted a request"));

			const toastSpy = vi.spyOn(toasts, "success");

			render(
				<FormProvider {...form.current}>
					<WalletOverviewStep />
				</FormProvider>,
			);

			await userEvent.click(screen.getByTestId("CopyOrDownload__download"));

			expect(toastSpy).not.toHaveBeenCalled();

			toastSpy.mockRestore();
			browserAccessMock.mockRestore();
		});

		it("should show error toast on error", async () => {
			const { result: form } = renderForm();

			const browserAccessMock = vi.spyOn(browserAccess, "fileSave").mockRejectedValue(new Error("Error"));

			const toastSpy = vi.spyOn(toasts, "error");

			render(
				<FormProvider {...form.current}>
					<WalletOverviewStep />
				</FormProvider>,
			);

			await userEvent.click(screen.getByTestId("CopyOrDownload__download"));

			await waitFor(() => {
				expect(toastSpy).toHaveBeenCalledWith(expect.stringMatching(/Could not save/));
			});

			toastSpy.mockRestore();
			browserAccessMock.mockRestore();
		});
	});
});
