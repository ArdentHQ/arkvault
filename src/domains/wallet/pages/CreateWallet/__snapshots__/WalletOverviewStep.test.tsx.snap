// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WalletOverviewStep > Render step > should render 1`] = `
<DocumentFragment>
  <section
    data-testid="CreateWallet__WalletOverviewStep"
  >
    <div
      class="flex items-end justify-between bg-theme-background hidden sm:block"
    >
      <div
        class="space-y-2"
      >
        <div
          class="flex items-center gap-3"
        >
          <h1
            class="mb-0 text-2xl leading-[29px]"
            data-testid="header__title"
          >
            Your Passphrase
          </h1>
        </div>
      </div>
    </div>
    <div
      class="mt-8 space-y-8"
    >
      <div
        class="flex flex-col overflow-hidden rounded-xl mt-6"
      >
        <div
          class="css-rzhg83"
        >
          <div
            class="css-v0ob3f"
            height="16"
            width="16"
          >
            <svg
              id="circle-exclamation-mark"
              viewBox="0 0 20 20"
              x="0"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              y="0"
            >
              <g
                fill="none"
                stroke="currentColor"
                stroke-width="2"
              >
                <circle
                  cx="10"
                  cy="10"
                  r="9"
                  stroke-miterlimit="10"
                />
                <path
                  d="M10 10.5v-5"
                  stroke-linecap="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="M10 14h0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </g>
            </svg>
          </div>
          <span>
            Warning
          </span>
        </div>
        <div
          class="css-trjt6n"
        >
          You are responsible for storing and protecting this mnemonic passphrase offline. ARK Vault cannot reveal this to you at a later time. If you lose this mnemonic passphrase, you will lose your funds.
        </div>
      </div>
      <ul
        class="grid grid-cols-2 gap-x-3 gap-y-5 sm:grid-cols-4"
      >
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            1
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            1
          </div>
          <div
            class="sm:text-md text-sm"
          >
            skin
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            2
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            2
          </div>
          <div
            class="sm:text-md text-sm"
          >
            fortune
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            3
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            3
          </div>
          <div
            class="sm:text-md text-sm"
          >
            security
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            4
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            4
          </div>
          <div
            class="sm:text-md text-sm"
          >
            mom
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            5
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            5
          </div>
          <div
            class="sm:text-md text-sm"
          >
            coin
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            6
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            6
          </div>
          <div
            class="sm:text-md text-sm"
          >
            hurdle
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            7
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            7
          </div>
          <div
            class="sm:text-md text-sm"
          >
            click
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            8
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            8
          </div>
          <div
            class="sm:text-md text-sm"
          >
            emotion
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            9
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            9
          </div>
          <div
            class="sm:text-md text-sm"
          >
            heart
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            10
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            10
          </div>
          <div
            class="sm:text-md text-sm"
          >
            brisk
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            11
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            11
          </div>
          <div
            class="sm:text-md text-sm"
          >
            exact
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            12
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            12
          </div>
          <div
            class="sm:text-md text-sm"
          >
            rather
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            13
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            13
          </div>
          <div
            class="sm:text-md text-sm"
          >
            code
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            14
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            14
          </div>
          <div
            class="sm:text-md text-sm"
          >
            feature
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            15
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            15
          </div>
          <div
            class="sm:text-md text-sm"
          >
            era
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            16
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            16
          </div>
          <div
            class="sm:text-md text-sm"
          >
            leopard
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            17
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            17
          </div>
          <div
            class="sm:text-md text-sm"
          >
            grocery
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            18
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            18
          </div>
          <div
            class="sm:text-md text-sm"
          >
            tide
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            19
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            19
          </div>
          <div
            class="sm:text-md text-sm"
          >
            gift
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            20
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            20
          </div>
          <div
            class="sm:text-md text-sm"
          >
            power
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            21
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            21
          </div>
          <div
            class="sm:text-md text-sm"
          >
            lawsuit
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            22
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            22
          </div>
          <div
            class="sm:text-md text-sm"
          >
            sight
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            23
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            23
          </div>
          <div
            class="sm:text-md text-sm"
          >
            vehicle
          </div>
        </li>
        <li
          class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
          data-testid="MnemonicList__item"
        >
          <span
            class="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
          >
            24
          </span>
          <div
            class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
          >
            24
          </div>
          <div
            class="sm:text-md text-sm"
          >
            coin
          </div>
        </li>
      </ul>
      <div
        class="-mx-8 space-y-2 bg-theme-secondary-100 p-6 text-center dark:bg-black sm:mx-0 sm:rounded-xl sm:text-left"
      >
        <span
          class="text-lg font-semibold text-theme-secondary-text"
        >
          Copy or Download Mnemonic
        </span>
        <p
          class="text-sm text-theme-secondary-500"
        >
          You can copy or download your mnemonic, but store it safely.
        </p>
        <div
          class="flex items-center justify-center space-x-3 sm:justify-start"
        >
          <button
            class="ring-focus relative focus:outline-none"
            data-ring-focus-margin="-m-1"
            data-testid="clipboard-icon__wrapper"
            type="button"
          >
            <div
              class="flex items-center space-x-2 font-semibold link"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg
                  id="copy"
                  viewBox="0 0 20 20"
                  x="0"
                  xml:space="preserve"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  >
                    <path
                      d="M7 17.765C7 18.485 7.496 19 8.09 19h-.099 8.82c.694 0 1.189-.618 1.189-1.235V6.235C18 5.515 17.405 5 16.81 5h0-6.242c-.298 0-.595.103-.892.309l-2.379 2.47C7 8.09 7 8.397 7 8.706v9.059z"
                    />
                    <path
                      d="M13 5V2.235C13 1.515 12.405 1 11.81 1h0-6.242c-.298 0-.595.103-.892.309l-2.379 2.47C2 4.09 2 4.397 2 4.706v9.059C2 14.485 2.496 15 3.09 15h-.099H7"
                    />
                  </g>
                </svg>
              </div>
              <span>
                Copy
              </span>
            </div>
          </button>
          <div
            class="border-theme-secondary-300 dark:border-theme-secondary-800 css-1aie31u"
            type="vertical"
          />
          <button
            class="flex items-center space-x-2 font-semibold link"
            data-testid="CopyOrDownload__download"
            type="button"
          >
            <div
              class="css-v0ob3f"
              height="16"
              width="16"
            >
              <svg
                id="arrow-down-bracket"
                viewBox="0 0 20 20"
                x="0"
                xml:space="preserve"
                xmlns="http://www.w3.org/2000/svg"
                y="0"
              >
                <path
                  d="M19 13.4v1.9c0 2.1-1.7 3.7-3.7 3.7H4.7c-2 0-3.7-1.7-3.7-3.7v-1.9M9.95 2v11.1m4.7-4.7l-4.7 4.7-4.6-4.7"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <span>
              Download
            </span>
          </button>
        </div>
      </div>
      <div
        class="flex w-full flex-col space-y-2"
      >
        <div
          class="flex items-center justify-between space-x-5"
        >
          <span
            class="font-bold text-theme-secondary-text"
          >
            Use Wallet Encryption
          </span>
          <span
            data-testid="CreateWallet__encryption"
          >
            <label
              class="css-xklln0"
            >
              <input
                class="css-1r71f0z toggle-input"
                data-testid="CreateWallet__encryption-toggle"
                type="checkbox"
              />
              <div
                class="css-d8rk4r toggle-handle"
              >
                <span
                  class="css-st84t0"
                />
              </div>
            </label>
          </span>
        </div>
        <span
          class="mr-12 text-sm text-theme-secondary-500"
        >
          Set an encryption password to use in place of your mnemonic passphrase. Note that you must still record and keep your mnemonic passphrase safe as losing this will result in you losing all access to your funds.
        </span>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
